File: src/HH20Backend/HH20Core/Models/Document.cs
using System;
using Dapper.Contrib.Extensions;

namespace HH20Core.Models
{
    [Table("Document")]
    public class Document
    {
        [ExplicitKey]
        public Guid Id { get; set; }
        public DateTimeOffset UploadedAt { get; set; }
        public DateTimeOffset DocumentTime { get; set; }
        public Guid UserId { get; set; }
        public Guid ActivityId { get; set; }
        public DocumentType DocumentType { get; set; }
        public string Filename { get; set; }
        public string Mimetype { get; set; }
        public string BlobIdentifier { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
    }
}

File: src/HH20Mobile/HealthHack/Server/Model/Document.swift
//
//  Document.swift
//  HealthHack
//
//  Created by Anatoli Benke on 07.11.20.
//  Copyright © 2020 Anatoli Benke. All rights reserved.
//

import Foundation



struct Document: Codable {
    let id: String
    let uploadedAt: String
    let documentTime: String
    let documentType: Int
    let filename: String?
    let mimetype: String
    let title: String
    let description: String?
    
    
    
    func getDocumentTimeDate() -> Date {
        return DateHelper.removeTimeStamp(fromDate: DateHelper.dateFromDateTimeString(string: documentTime)!)
    }
}


struct DocumentSection: Codable {
    let date: Date
    let documents: [Document]
}
